{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO # Biopython for fast parsing of FASTA data\n",
    "import pandas as pd # pandas for dataframes to organize outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def complement(seq):\n",
    "    \"\"\"\n",
    "    Takes in a sequence and output the complement of that sequence. Note, this \n",
    "    does not reverse the sequence so the output is not the reverse complement.\n",
    "\n",
    "    seq: String. The sequence to get the complement of.\n",
    "    Return: String. The complement of the input sequence.\n",
    "    \n",
    "    \"\"\"\n",
    "    comp = {'A':'T', 'T':'A', 'G':'C', 'C':'G'}\n",
    "    comp_seq = \"\".join(comp.get(base, base) for base in seq)\n",
    "    return comp_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>chromosome_end</th>\n",
       "      <th>chromosome_strand</th>\n",
       "      <th>assembly_version</th>\n",
       "      <th>reference_genome_allele</th>\n",
       "      <th>mutated_to_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6</td>\n",
       "      <td>107956302</td>\n",
       "      <td>107956302</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>107956302</td>\n",
       "      <td>107956302</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6</td>\n",
       "      <td>107956312</td>\n",
       "      <td>107956312</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>107956312</td>\n",
       "      <td>107956312</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>X</td>\n",
       "      <td>76920372</td>\n",
       "      <td>76920372</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>X</td>\n",
       "      <td>76920372</td>\n",
       "      <td>76920372</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>229781704</td>\n",
       "      <td>229781704</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>229781704</td>\n",
       "      <td>229781704</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>17</td>\n",
       "      <td>56738869</td>\n",
       "      <td>56738869</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>17</td>\n",
       "      <td>56738869</td>\n",
       "      <td>56738869</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  chromosome_start  chromosome_end chromosome_strand  \\\n",
       "0          6         107956302       107956302                 +   \n",
       "1          6         107956302       107956302                 -   \n",
       "2          6         107956312       107956312                 +   \n",
       "3          6         107956312       107956312                 -   \n",
       "4          X          76920372        76920372                 +   \n",
       "5          X          76920372        76920372                 -   \n",
       "6          1         229781704       229781704                 +   \n",
       "7          1         229781704       229781704                 -   \n",
       "8         17          56738869        56738869                 +   \n",
       "9         17          56738869        56738869                 -   \n",
       "\n",
       "  assembly_version reference_genome_allele mutated_to_allele  \n",
       "0           GRCh37                       G                 A  \n",
       "1           GRCh37                       C                 T  \n",
       "2           GRCh37                       A                 G  \n",
       "3           GRCh37                       T                 C  \n",
       "4           GRCh37                       G                 A  \n",
       "5           GRCh37                       C                 T  \n",
       "6           GRCh37                       C                 T  \n",
       "7           GRCh37                       G                 A  \n",
       "8           GRCh37                       G                 A  \n",
       "9           GRCh37                       C                 T  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load in the text file with the mutation targets\n",
    "mutations = pd.read_csv('somatic_mutation_data.txt', sep='\\t')\n",
    "mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "sequences = [] # Store the target sequences\n",
    "\n",
    "for ind in range(len(mutations)):\n",
    "    # Load information from mutation data table\n",
    "    chr = 'chr' + mutations['chromosome'][ind] # Chromosome number of target\n",
    "    loc = mutations['chromosome_start'][ind] - 1 # Location of target\n",
    "    sense = mutations['chromosome_strand'][ind].strip() # Strand information\n",
    "    ref = mutations['reference_genome_allele'][ind] # Expected reference base\n",
    "    \n",
    "    for sequence in SeqIO.parse(open('hg19.fa'), 'fasta'):\n",
    "        if sequence.id == chr:\n",
    "            seq = str(sequence.seq).upper()\n",
    "            break\n",
    "    \n",
    "    # Check if the base at location is the same as the expected reference\n",
    "    if seq[loc] != ref and sense == '+':\n",
    "        raise Exception(\"Error in parsing sequence\")\n",
    "\n",
    "    # Get the 200bp sequence\n",
    "    snp_seq = seq[loc-100:loc+100]\n",
    "\n",
    "    # Check to see if the complement sequence needs to be calculated and append\n",
    "    if sense == '+':\n",
    "        sequences.append(snp_seq)\n",
    "    else:\n",
    "        sequences.append(complement(snp_seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the counts of A, C, G, T in each extracted sequences\n",
    "count_A = [sect.count('A') for sect in sequences]\n",
    "count_C = [sect.count('C') for sect in sequences]\n",
    "count_G = [sect.count('G') for sect in sequences]\n",
    "count_T = [sect.count('T') for sect in sequences]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[200, 200, 200, 200, 200, 200, 200, 200, 200, 200]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sanity check to make sure all the values add back up to 200 so that there wiil\n",
    "#  not be any double or missing count\n",
    "check_sum = [sum(x) for x in zip(count_A, count_C, count_G, count_T)]\n",
    "for count, total in enumerate(check_sum):\n",
    "    if total != 200:\n",
    "        raise Exception(f\"Mutation {count} individual single base total is not \n",
    "                        200\")\n",
    "check_sum"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Get the 2-Gram counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_AA = [sect.count('AA') for sect in sequences]\n",
    "count_AC = [sect.count('AC') for sect in sequences]\n",
    "count_AG = [sect.count('AG') for sect in sequences]\n",
    "count_AT = [sect.count('AT') for sect in sequences]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_CA = [sect.count('CA') for sect in sequences]\n",
    "count_CC = [sect.count('CC') for sect in sequences]\n",
    "count_CG = [sect.count('CG') for sect in sequences]\n",
    "count_CT = [sect.count('CT') for sect in sequences]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_GA = [sect.count('GA') for sect in sequences]\n",
    "count_GC = [sect.count('GC') for sect in sequences]\n",
    "count_GG = [sect.count('GG') for sect in sequences]\n",
    "count_GT = [sect.count('GT') for sect in sequences]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_TA = [sect.count('TA') for sect in sequences]\n",
    "count_TC = [sect.count('TC') for sect in sequences]\n",
    "count_TG = [sect.count('TG') for sect in sequences]\n",
    "count_TT = [sect.count('TT') for sect in sequences]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Put all the results back together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutations['Sequence'] = sequences\n",
    "mutations['A count'] = count_A\n",
    "mutations['C count'] = count_C\n",
    "mutations['G count'] = count_G\n",
    "mutations['T count'] = count_T\n",
    "mutations['AA count'] = count_AA\n",
    "mutations['AC count'] = count_AC\n",
    "mutations['AG count'] = count_AG\n",
    "mutations['AT count'] = count_AT\n",
    "mutations['CA count'] = count_CA\n",
    "mutations['CC count'] = count_CC\n",
    "mutations['CG count'] = count_CG\n",
    "mutations['CT count'] = count_CT\n",
    "mutations['GA count'] = count_GA\n",
    "mutations['GC count'] = count_GC\n",
    "mutations['GG count'] = count_GG\n",
    "mutations['GT count'] = count_GT\n",
    "mutations['TA count'] = count_TA\n",
    "mutations['TC count'] = count_TC\n",
    "mutations['TG count'] = count_TG\n",
    "mutations['TT count'] = count_TT"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>chromosome_end</th>\n",
       "      <th>chromosome_strand</th>\n",
       "      <th>assembly_version</th>\n",
       "      <th>reference_genome_allele</th>\n",
       "      <th>mutated_to_allele</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>A count</th>\n",
       "      <th>C count</th>\n",
       "      <th>...</th>\n",
       "      <th>CG count</th>\n",
       "      <th>CT count</th>\n",
       "      <th>GA count</th>\n",
       "      <th>GC count</th>\n",
       "      <th>GG count</th>\n",
       "      <th>GT count</th>\n",
       "      <th>TA count</th>\n",
       "      <th>TC count</th>\n",
       "      <th>TG count</th>\n",
       "      <th>TT count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6</td>\n",
       "      <td>107956302</td>\n",
       "      <td>107956302</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>GGCCTGCAACGTCATCGTGAACGGCACGCGCGGCGCCGCCGCCGAG...</td>\n",
       "      <td>35</td>\n",
       "      <td>74</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>33</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6</td>\n",
       "      <td>107956302</td>\n",
       "      <td>107956302</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CCGGACGTTGCAGTAGCACTTGCCGTGCGCGCCGCGGCGGCGGCTC...</td>\n",
       "      <td>21</td>\n",
       "      <td>70</td>\n",
       "      <td>...</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>29</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6</td>\n",
       "      <td>107956312</td>\n",
       "      <td>107956312</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>GTCATCGTGAACGGCACGCGCGGCGCCGCCGCCGAGGGCGCTAAGA...</td>\n",
       "      <td>36</td>\n",
       "      <td>73</td>\n",
       "      <td>...</td>\n",
       "      <td>29</td>\n",
       "      <td>10</td>\n",
       "      <td>15</td>\n",
       "      <td>33</td>\n",
       "      <td>14</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>9</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6</td>\n",
       "      <td>107956312</td>\n",
       "      <td>107956312</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>CAGTAGCACTTGCCGTGCGCGCCGCGGCGGCGGCTCCCGCGATTCT...</td>\n",
       "      <td>20</td>\n",
       "      <td>71</td>\n",
       "      <td>...</td>\n",
       "      <td>33</td>\n",
       "      <td>15</td>\n",
       "      <td>10</td>\n",
       "      <td>29</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>X</td>\n",
       "      <td>76920372</td>\n",
       "      <td>76920372</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>AGAGTAAAAAACAATAAAAAAAGAACTATATATCTGGGGGTGAAAT...</td>\n",
       "      <td>97</td>\n",
       "      <td>25</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>24</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>X</td>\n",
       "      <td>76920372</td>\n",
       "      <td>76920372</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>TCTCATTTTTTGTTATTTTTTTCTTGATATATAGACCCCCACTTTA...</td>\n",
       "      <td>58</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>21</td>\n",
       "      <td>8</td>\n",
       "      <td>14</td>\n",
       "      <td>97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>229781704</td>\n",
       "      <td>229781704</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>AGGCTGTTGTGTCAGCTGTTACACTGCTGAGGCTGCTACTGAACTG...</td>\n",
       "      <td>38</td>\n",
       "      <td>56</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>7</td>\n",
       "      <td>19</td>\n",
       "      <td>7</td>\n",
       "      <td>15</td>\n",
       "      <td>23</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>229781704</td>\n",
       "      <td>229781704</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>TCCGACAACACAGTCGACAATGTGACGACTCCGACGATGACTTGAC...</td>\n",
       "      <td>54</td>\n",
       "      <td>52</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>24</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>9</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>17</td>\n",
       "      <td>56738869</td>\n",
       "      <td>56738869</td>\n",
       "      <td>+</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>GAACAAGTCAAGCTGCGATAACATTTTGGCAACCTTGTCAAGGAGG...</td>\n",
       "      <td>42</td>\n",
       "      <td>53</td>\n",
       "      <td>...</td>\n",
       "      <td>14</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>11</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>17</td>\n",
       "      <td>56738869</td>\n",
       "      <td>56738869</td>\n",
       "      <td>-</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>CTTGTTCAGTTCGACGCTATTGTAAAACCGTTGGAACAGTTCCTCC...</td>\n",
       "      <td>46</td>\n",
       "      <td>59</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>16</td>\n",
       "      <td>14</td>\n",
       "      <td>9</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "      <td>13</td>\n",
       "      <td>10</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  chromosome  chromosome_start  chromosome_end chromosome_strand  \\\n",
       "0          6         107956302       107956302                 +   \n",
       "1          6         107956302       107956302                 -   \n",
       "2          6         107956312       107956312                 +   \n",
       "3          6         107956312       107956312                 -   \n",
       "4          X          76920372        76920372                 +   \n",
       "5          X          76920372        76920372                 -   \n",
       "6          1         229781704       229781704                 +   \n",
       "7          1         229781704       229781704                 -   \n",
       "8         17          56738869        56738869                 +   \n",
       "9         17          56738869        56738869                 -   \n",
       "\n",
       "  assembly_version reference_genome_allele mutated_to_allele  \\\n",
       "0           GRCh37                       G                 A   \n",
       "1           GRCh37                       C                 T   \n",
       "2           GRCh37                       A                 G   \n",
       "3           GRCh37                       T                 C   \n",
       "4           GRCh37                       G                 A   \n",
       "5           GRCh37                       C                 T   \n",
       "6           GRCh37                       C                 T   \n",
       "7           GRCh37                       G                 A   \n",
       "8           GRCh37                       G                 A   \n",
       "9           GRCh37                       C                 T   \n",
       "\n",
       "                                            Sequence  A count  C count  ...  \\\n",
       "0  GGCCTGCAACGTCATCGTGAACGGCACGCGCGGCGCCGCCGCCGAG...       35       74  ...   \n",
       "1  CCGGACGTTGCAGTAGCACTTGCCGTGCGCGCCGCGGCGGCGGCTC...       21       70  ...   \n",
       "2  GTCATCGTGAACGGCACGCGCGGCGCCGCCGCCGAGGGCGCTAAGA...       36       73  ...   \n",
       "3  CAGTAGCACTTGCCGTGCGCGCCGCGGCGGCGGCTCCCGCGATTCT...       20       71  ...   \n",
       "4  AGAGTAAAAAACAATAAAAAAAGAACTATATATCTGGGGGTGAAAT...       97       25  ...   \n",
       "5  TCTCATTTTTTGTTATTTTTTTCTTGATATATAGACCCCCACTTTA...       58       20  ...   \n",
       "6  AGGCTGTTGTGTCAGCTGTTACACTGCTGAGGCTGCTACTGAACTG...       38       56  ...   \n",
       "7  TCCGACAACACAGTCGACAATGTGACGACTCCGACGATGACTTGAC...       54       52  ...   \n",
       "8  GAACAAGTCAAGCTGCGATAACATTTTGGCAACCTTGTCAAGGAGG...       42       53  ...   \n",
       "9  CTTGTTCAGTTCGACGCTATTGTAAAACCGTTGGAACAGTTCCTCC...       46       59  ...   \n",
       "\n",
       "   CG count  CT count  GA count  GC count  GG count  GT count  TA count  \\\n",
       "0        29        11        14        33        13         7         3   \n",
       "1        33        14        11        29        22         8         2   \n",
       "2        29        10        15        33        14         7         3   \n",
       "3        33        15        10        29        22         8         2   \n",
       "4         0        10         7         3         2         6        24   \n",
       "5         3         7        10         0         4        11        21   \n",
       "6         2        24        11        14         7        19         7   \n",
       "7        14        11        24         2        12        11         2   \n",
       "8        14        16        14        16        16         9         2   \n",
       "9        16        14        16        14         9        13         5   \n",
       "\n",
       "   TC count  TG count  TT count  \n",
       "0         7        10        21  \n",
       "1        15         8        35  \n",
       "2         7         9        20  \n",
       "3        16         7        36  \n",
       "4         4         8        58  \n",
       "5         8        14        97  \n",
       "6        15        23        54  \n",
       "7        19         9        38  \n",
       "8        17        11        46  \n",
       "9        13        10        42  \n",
       "\n",
       "[10 rows x 28 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutations"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
